internal/authorizer/authorizer.go

```diff
@@ -90,7 +91,15 @@ func (a *Authorizer) Authorize(
 
 func (a *Authorizer) authorizeInner(user string, groups []string, verb, resource, resourceName, apiGroup string, namespaces []string, metadataOnly bool) (types.DataResponseV1, error) {
 	// check if user has cluster-wide access
-	clusterAllow, err := a.client.SubjectAccessReview(user, groups, verb, resource, resourceName, apiGroup, "")
+	var (
+		clusterAllow bool
+		err          error
+	)
+	if a.ssar {
+		clusterAllow, err = a.client.SelfSubjectAccessReview(verb, resource, resourceName, apiGroup, "")
+	} else {
+		clusterAllow, err = a.client.SubjectAccessReview(user, groups, verb, resource, resourceName, apiGroup, "")
+	}
```

Comment:
Looking at the issue that we also need the distinction "do we do SSAR or SAR" in the client (because it decides which credentials will be used) I think I'd push this logic down to the client completely and have a single "access review" function in the client interface that does either SAR or SSAR. What do you think?

Reply:
Great idea!